pulsar-broker/src/test/java/org/apache/pulsar/client/api/TopicReaderTest.java

```diff
@@ -86,6 +87,33 @@ public void testSimpleReader() throws Exception {
         producer.close();
     }
 
+    @Test
+    public void testReaderAtSelectedTimestamp() throws Exception {
+         Producer<byte[]> producer = pulsarClient.newProducer().topic("persistent://my-property/my-ns/testReaderAtSelectedTimestamp")
+                .create();
+         for (int i = 0; i < 10; i++) {
+            String message = "my-message-" + i;
+            producer.send(message.getBytes());
+            Thread.sleep(1000);
```

Comment:
why do you add `sleep` here?

Reply:
My idea was to use the sleep so that we could space the publish times of the messages 1000ms apart (although that might or might have not worked).